lib/fs.js

```diff
@@ -1452,7 +1469,7 @@ function readdirSyncRecursive(basePath, options) {
       for (let i = 0; i < length; i++) {
         const dirent = getDirent(path, readdirResult[0][i], readdirResult[1][i]);
         ArrayPrototypePush(readdirResults, dirent);
-        if (dirent.isDirectory()) {
+        if (dirent.isDirectory() && depthCheck(readdirResult[0][i])) {
```

Comment:
Isn't `readdirResult[0][1]` potentially a `Buffer`?

Reply:
According to the comment immediately preceding this logic readdirResult is an array of two arrays where the first is the name of file system artifacts and the second is the type.  Can a file system item have a buffer as a name?  If so I will add a `.toString()` to ensure coercion to a string unless there is a better way to handle this.  The value that is passed in will be converted to an absolute path.

Reply:
> Can a file system item have a buffer as a name?

Yes, if you set the `encoding` option to `'buffer'`.

> If so I will add a `.toString()` to ensure coercion to a string

`toString()` will decode the `Buffer` as a sequence of UTF-8 code points. However, as far as I can tell, there is no guarantee that the name will actually be a UTF-8 string.

> unless there is a better way to handle this.

I am not aware of any way to handle this. The `node:fs` module used to be a nice low-level interface for well-defined syscalls that would let the operating system handle paths.

Reply:
In that case it sounds like there is not a good way to account for that scenario, so I can address this in the documentation and ensure it does not result in execution errors.